packages/hackmode/lib/index.js

```diff
@@ -28,6 +28,15 @@ class N3hHackMode extends AsyncClass {
     this._memory = {}
     this._peerBook = {}
 
+    // Book: Array of entryAddress (or metaId) per dna
+    this._publishedEntryBook = {}
+    this._storedEntryBook = {}
+    this._publishedMetaBook = {}
+    this._storedMetaBook = {}
+
+    this._requestBook = {}
```

Comment:
especially if you are removing items, [Maps](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) are more efficient, and the api is more functional.

```javascript
this._requestBook = new Map()

this._requestBook.has(key) // boolean
this._requestBook.delete(key)
```

Reply:
done